controllers/secret_controller.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/go-logr/logr"
+
+	"github.com/giantswarm/capa-iam-operator/pkg/awsclient"
+	"github.com/giantswarm/capa-iam-operator/pkg/iam"
+)
+
+const (
+	IRSASecretSuffix = "irsa-cloudfront"
+)
+
+// SecretReconciler reconciles a Secret object
+type SecretReconciler struct {
+	client.Client
+	EnableKiamRole bool
+	Log            logr.Logger
+	Scheme         *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=core,resources=secrets/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=core,resources=secrets/finalizers,verbs=update
+
+func (r *SecretReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.TODO()
+
+	logger := r.Log.WithValues("namespace", req.Namespace, "secret", req.Name)
+
+	if r.EnableKiamRole {
```

Comment:
atm we might need both Kiam and irsa to be working before we can fully switch

Reply:
Do you mean have them both working concurrently on the cluster or to just have the ability to choose between the two?

Reply:
we don't need ability to switch between two but we need to be able to support both at the same time because atm we only use KIAM and not IRSA, if  this is merged then the current state of MCs can be broken until we upgrade `default-apps-aws`  and I would like to avoid broken state 
 
 once all clusters are using IRSA only and can remove any kiam dependencies from the operator

Reply:
Thank you for the clarification :). Will make changes to the PR